### PR TITLE
fix token to token swap

### PR DESCRIPTION
**Pull request by ben2x4**
We are having issues with pool that do not include Juno. This PR makes token to token swap route finding and price querying work for all pools.

### CODE REVIEW

File: features/swap/hooks/useTokenSwap.tsx [hunk 1]
```diff
@@ -110,8 +110,8 @@ export const useTokenSwap = ({
         slippage,
         senderAddress: address,
         tokenA,
-        swapAddress: passThroughPool.inputPool.swap_address,
-        outputSwapAddress: passThroughPool.outputPool.swap_address,
+        swap: passThroughPool.inputPool,
+        outputSwap: passThroughPool.outputPool,
```
Comment (sashimi36) [inline features/swap/hooks/useTokenSwap.tsx]: Wondering if being more precise on naming can bring more clarity on what's being passed here. These are pools right?

File: queries/useQueryMatchingPoolForSwap/util/selectEligiblePoolsForTokenToTokenSwap.ts [hunk 2]
```diff
@@ -45,29 +44,31 @@ export function selectEligiblePoolsForTokenToTokenSwap({
...
+          tokenA.symbol === poolAssetA.symbol ? poolAssetB : poolAssetA
 
-        const passThroughSwapInputPool = isValidSwapInputPool
-          ? pool
-          : selectPoolByTokenPair(poolsList, poolAssetA, tokenA)
-
-        const passThroughSwapOutputPool = isValidSwapOutputPool
-          ? pool
-          : selectPoolByTokenPair(poolsList, poolAssetA, tokenB)
+        let passThroughSwapOutputPool = poolsList.find(
```
Comment (sashimi36) [inline queries/useQueryMatchingPoolForSwap/util/selectEligiblePoolsForTokenToTokenSwap.ts]: Think this codebase style largely follows `const` for variables that aren't gonna change and `let` for variables that are more flexible in that regard.

File: queries/useQueryMatchingPoolForSwap/util/selectEligiblePoolsForTokenToTokenSwap.ts [hunk 3]
```diff
@@ -45,29 +44,31 @@ export function selectEligiblePoolsForTokenToTokenSwap({
...
+          ({ pool_assets: [assetA, assetB] }) => {
```
Comment (sashimi36) [inline queries/useQueryMatchingPoolForSwap/util/selectEligiblePoolsForTokenToTokenSwap.ts:56]: Loving the destructuring here

File: services/swap/price.ts [hunk 4]
```diff
@@ -48,29 +49,48 @@ export const getToken2ForToken1Price = async ({
...
+            swapAddress: swap.swap_address,
+            client,
+          })
 
-    return getToken1ForToken2Price({
-      nativeAmount: nativePrice,
-      swapAddress: outputSwapAddress,
-      client: client,
-    })
+    let result =
```
Comment (sashimi36) [inline services/swap/price.ts]: Nice! Looks cleaner this way. Can also just return this entire condition out the function and skip this assignment

**Review by sashimi36 — COMMENTED** [review]
Leaving a few code style suggestions otherwise looks great.

FYI didn't test if this works

**Review by sashimi36 — CHANGES_REQUESTED** [review]
@ben2x4 ok sounds like some of the pricing queries usage instances weren't updated accordingly. Check the logs out, can also do `yarn typecheck` locally.

By the way, would be great to merge #250 first. It includes one of the typescript fixes you might need here.

File: services/swap/price.ts (hunk 4 above)
Comment (ben2x4) [reply services/swap/price.ts]: good catch, was doing the assignment becuase i was logging the variable at one point

**Comment by ben2x4** [comment]
> @ben2x4 ok sounds like some of the pricing queries usage instances weren't updated accordingly. Check the logs out, can also do `yarn typecheck` locally.
> 
> By the way, would be great to merge #250 first. It includes one of the typescript fixes you might need here.

this is fixed, the query that was giving us issues is actually no longer used in the codebase so i deleted it.

File: queries/tokenToTokenPriceQuery.ts [hunk 5]
```diff
@@ -80,8 +80,10 @@ export async function tokenToTokenPriceQueryWithPools({
       pricingQueries.push(
         getTokenForTokenPrice({
           tokenAmount: convertedTokenAmount,
-          swapAddress: passThroughPool.inputPool.swap_address,
-          outputSwapAddress: passThroughPool.outputPool.swap_address,
+          tokenA,
+          tokenB,
+          swap: passThroughPool.inputPool,
+          outputSwap: passThroughPool.outputPool,
```
Comment (sashimi36) [inline queries/tokenToTokenPriceQuery.ts]: Think it's best to keep it `swap`/`outputSwap` or would rather make it more consistent like you do it in `useTokenSwap`?

File: services/swap/passThroughTokenSwap.ts [hunk 6]
```diff
@@ -13,30 +13,33 @@ type PassThroughTokenSwapArgs = {
...
   client: SigningCosmWasmClient
 }
 
 export const passThroughTokenSwap = async ({
   tokenAmount,
   tokenA,
-  outputSwapAddress,
-  swapAddress,
+  outputPool: outputSwap,
+  inputPool: swap,
```
Comment (sashimi36) [inline services/swap/passThroughTokenSwap.ts]: Sort of minuscule- why renaming it to different variables if can just use whatever you pass in here?

File: services/swap/price.ts [hunk 7]
```diff
@@ -48,29 +49,46 @@ export const getToken2ForToken1Price = async ({
...
 }
 
 export const getTokenForTokenPrice = async ({
   tokenAmount,
-  swapAddress,
-  outputSwapAddress,
+  tokenA,
+  tokenB,
+  swap,
+  outputSwap,
```
Comment (sashimi36) [inline services/swap/price.ts]: Same as `swap`/`outputSwap` thing above

**Review by sashimi36 — COMMENTED** [review]
Great stuff @ben2x4. Leaving a few remarks on the consistency otherwise really great to see how you just nail frontend architecture work like this.

> > @ben2x4 ok sounds like some of the pricing queries usage instances weren't updated accordingly. Check the logs out, can also do `yarn typecheck` locally.
> > By the way, would be great to merge #250 first. It includes one of the typescript fixes you might need here.
> 
> this is fixed, the query that was giving us issues is actually no longer used in the codebase so i deleted it.

Cool, sounds like the build is still failing though? Might be because we're still passing output/input swap addresses where the query expects to receive the entire pool data

**Review by sashimi36 — APPROVED** [review]
Code looks great.

Confirming the pricing math works as expected. 

🥇

**Pull request merged**